pkg/webhook/policy.go

```diff
@@ -501,7 +508,73 @@ func (h *validationHandler) reviewRequest(ctx context.Context, req *admission.Re
 		}
 	}
 
+	review, err := h.createReview(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create augmentedReview: %s", err)
+	}
+
+	// Convert the request's generator resource to unstructured for expansion
+	obj := &unstructured.Unstructured{}
+	if _, _, err := deserializer.Decode(req.Object.Raw, nil, obj); err != nil {
+		return nil, fmt.Errorf("error decoding generator resource %s: %s", req.Name, err)
+	}
+	obj.SetNamespace(req.Namespace)
+	obj.SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Group:   req.Kind.Group,
+			Version: req.Kind.Version,
+			Kind:    req.Kind.Kind,
+		})
+
+	// Expand the generator and apply mutators to the resultant resources
+	base := &mutationtypes.Mutable{
+		Object:    obj,
+		Namespace: review.Namespace,
+		Username:  req.AdmissionRequest.UserInfo.Username,
+		Source:    mutationtypes.SourceTypeGenerated,
+	}
+	resultants, err := h.expansionSystem.Expand(base)
+	if err != nil {
+		return nil, fmt.Errorf("error expanding generator: %s", err)
+	}
+
 	trace, dump := h.tracingLevel(ctx, req)
+	resp, err := h.review(ctx, review, trace, dump)
+	if err != nil {
+		return nil, fmt.Errorf("failed to review generator resource %s: %s", req.Name, err)
+	}
+
+	var resultantResps []*rtypes.Responses
+	for _, res := range resultants {
+		resp, err := h.review(ctx, res, trace, dump)
```

Comment:
This will need to be an augmented review so that namespace is sideloaded (if necessary), and also the fact that it's generated will need to be inserted so that the matcher can be evaluated.